tests/shakedown/shakedown/clients/authentication.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import toml
+
+from functools import lru_cache
+from os import environ, path
+from shakedown import http
+from shakedown.dcos import gen_url
+from shakedown.errors import DCOSException
+
+
+logger = logging.getLogger(__name__)
+
+
+@lru_cache
+def read_config():
+    """ Read configuration options from ~/.shakedown (if exists)
+        :return: a dict of arguments
+        :rtype: dict
+    """
+    configfile = path.expanduser('~/.shakedown')
+    args = dict()
+    if path.isfile(configfile):
+        with open(configfile, 'r') as f:
+            config = toml.loads(f.read())
+        for key in config:
+            param = key.replace('-', '_')
+            args[param] = config[key]
+    return args
+
+
+def authenticate(username, password):
+    """Authenticate with a DC/OS cluster and return an ACS token.
+    return: ACS token
```

Comment:
Terminology:

Log in to the cluster to obtain a DC/OS authentication token which can subsequently be used to authenticate individual HTTP requests.